### PR TITLE
Trim whitespace instead of removing spaces from the variables name

### PR DESCRIPTION
**Pull request by adeas31**
### Related Issues

#12235

### CODE REVIEW

**Comment by perost** [comment]
I'm probably not the right person to review this since I have no clue why the spaces were removed in the first place. Some test cases added to the test suite would be nice though, assuming we don't already have some.

**Review by perost — APPROVED** [review]

File: OMCompiler/SimulationRuntime/c/util/read_matlab4.c [hunk 1]
```diff
@@ -141,17 +141,20 @@ void omc_free_matlab4_reader(ModelicaMatReader *reader)
...
-            uj++;
-        }
-    }
+void trimWhitespace(char *str)
+{
+  char *end;
+  /* skip leading whitespace */
+  while (isspace(*str)) {
+    str++;
+  }
```
Comment (perost) [inline OMCompiler/SimulationRuntime/c/util/read_matlab4.c:150]: This just advances the local `str` pointer, it doesn't change the string. If the intent is to remove leading whitespace you need to actually change the string like the original function.

Comment (perost) [reply OMCompiler/SimulationRuntime/c/util/read_matlab4.c:150]:
```C
void trimWhitespace(char *str)
{
  char *start = str;
  char *end;
  size_t len;

  /* find first non-whitespace */
  while (isspace(*start)) {
    ++start;
  }

  /* find last non-whitespace */
  end = start + strlen(start) - 1;
  while (end > str && isspace(*end)) {
    --end;
  }

  len = end-start+1;

  /* shift the string to remove leading whitespace */
  if (start != str) memmove(str, start, len);

  /* terminate the string to remove trailing whitespace */
  *(str+len) = '\0';
}
```

Comment (adeas31) [reply OMCompiler/SimulationRuntime/c/util/read_matlab4.c]: Yes, all the reference files variables only have the trailing spaces so I forgot to remove the leading spaces :smile:. I updated and improved the code.

**Review by perost — APPROVED** [review]

**Pull request merged**